src/daemon/default_vm_image_vault.cpp

```diff
@@ -431,6 +431,8 @@ mp::VMImage mp::DefaultVMImageVault::fetch_image(const FetchType& fetch_type, co
         }
         catch (const AbortedDownloadException&)
```

Comment:
Ok, I've looked in to this more.  I've confirmed it doesn't regress the other issue, but since it didn't delete the in-flight future as the generic exception handler did, it caused the issue the bug this is solving.  Other than having a separate exception type, I don't see anything special about `AbortedDownloadException` any more.  I guess we can leave it in, but it doesn't really add anything either :shrug:

Reply:
OK, I think it makes sense to leave the `AbortedDownloadException` type around. But the catch itself may as well be removed now, right? 

I didn't quite understand why the catch was here before, but it appeared as if the exception type was introduced expressly for it (?) which led me to believe it was somehow important. If you're confident the catch is not needed anymore, I am fine with dropping it.

Reply:
Ok, I'd like to get rid of it too.  One question though.  Both the `catch` statements` re-throw, so if the following generic `catch(const std::exception&)` catches the `AbortedDownloadException`, will the re-throw throw the same exception?  I've yet to find proof if it does, but if you know, that would be great.

Reply:
Alright, I have confirmed the original exception is re-thrown since it was caught by reference.  So I will just remove this :slightly_smiling_face:

Reply:
Ok, I removed this handler.

Reply:
Yep, `throw;` (with no exception) throws the current exception, regardless of the type used in the `catch`. Note that's not the case if you say `throw e;` which would copy construct a new exception and incur in slicing (even if e was caught by reference).